### ISSUE TITLE
Default for server.tomcat.basedir shoudn't be /tmp

### ISSUE THREAD

**Issue by krzysztofslusarski** [spring-projects/spring-boot#25890]
The default for ```server.tomcat.basedir``` is located in temporary directory of OS. This may cause a functional issue in the application using JSP/Tags. 

Source code for a sample application [on GitHub](https://github.com/krzysztofslusarski/articles-spring-boot-jsp).

```java
@SpringBootApplication
public class JspApplication extends SpringBootServletInitializer {

    @Override
    protected SpringApplicationBuilder configure(SpringApplicationBuilder application) {
        return application.sources(JspApplication.class);
    }

    public static void main(String[] args) {
        SpringApplication.run(JspApplication.class, args);
    }

}
```

```java
@Controller
class JspController {

    @GetMapping("/")
    public String hello() {
        return "hello";
    }

    @GetMapping("/2")
    public String hello2() {
        return "hello2";
    }

}
```

... with the properties:

```properties
spring.mvc.view.prefix: /WEB-INF/jsp/
spring.mvc.view.suffix: .jsp
logging.level.root=DEBUG
```

Two JSP pages ```hello.jsp``` and ```hello2.jsp``` with the same content:

```jsp
<!DOCTYPE html>
<%@taglib prefix="inc" tagdir="/WEB-INF/tags" %>

<html lang="en">
<head>
    <meta charset="UTF-8">
</head>
<body>
Hello
<inc:included/>
</body>
</html>
```

... and one tag file `included.tag`:

```jsp
<%@tag pageEncoding="UTF-8" %>
<b>Included hello</b>
```

Step to reproduce an issue:
* Start the application
* Open ```http://localhost:8080/```
* Remove work directory: ```rm -fr /tmp/tomcat*```
* Open ```http://localhost:8080/2```

That will cause an error:

```
An error occurred at line: [10] in the jsp file: [/WEB-INF/jsp/hello2.jsp]
org.apache.jsp.tag.web.included_tag cannot be resolved to a type
7: </head>
8: <body>
9: Hello
10: <inc:included/>
11: </body>
12: </html>
```

Tomcat will not recompile deleted tag.

Temporary directory can be cleaned anytime, for example in ```CentOS``` it is cleaned by [tmpfiles.d](https://www.thegeekdiary.com/centos-rhel-67-why-the-files-in-tmp-directory-gets-deleted-periodically/)

**Comment by wilkinsona** [comment]
Thanks for the report. Using something other than `/tmp` has been [reconsidered before](https://github.com/spring-projects/spring-boot/issues/8459). I don't think anything's changed since then so we're unlikely to change the default.

That said, we used to have a note in the documentation about configuring tmpwatch, but it [was removed](https://github.com/spring-projects/spring-boot/commit/70eee612ff2a2b1e58cbcb18a4d46e464895c18a#diff-5a9b47517d10fa1ae24585a01dc8684d52cf1947e898b88ec0ee4bd9dc689199). Judging by what you have described above, that removal may have been rather hasty. We should explore if it's possible to configure Tomcat to recompile the deleted tag or if we need to reinstate the warning in a modified form.
